### PR TITLE
Feature/time trigger 2

### PR DESCRIPTION
**Pull request by kevinwallimann**
See #48 . This is a draft.

Missing:
- Separate thread pool for quartz jobs
- Integration test for TimeSensor

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/persistance/SensorRepository.scala [hunk 1]
```diff
@@ -17,11 +17,14 @@ package za.co.absa.hyperdrive.trigger.persistance
...
 import org.springframework.stereotype
 import za.co.absa.hyperdrive.trigger.models.Sensor
+import za.co.absa.hyperdrive.trigger.models.enums.SensorTypes
+
 import scala.concurrent.{ExecutionContext, Future}
 
 trait SensorRepository extends Repository {
   def getNewActiveSensors(idsToFilter: Seq[Long])(implicit ec: ExecutionContext): Future[Seq[Sensor]]
   def getInactiveSensors(ids: Seq[Long])(implicit ec: ExecutionContext): Future[Seq[Long]]
+  def getActiveTimeSensors()(implicit ec: ExecutionContext): Future[Seq[Sensor]]
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/persistance/SensorRepository.scala]: This method is not used anywhere.

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/sensors/Sensors.scala [hunk 2]
```diff
@@ -78,7 +81,12 @@ class Sensors @Inject()(eventProcessor: EventProcessor, sensorRepository: Sensor
 
           Try(new KafkaSensor(eventProcessor.eventProcessor, sensor.properties, executionContext)) match {
             case Success(s) => sensors.put(sensor.id, s)
-            case Failure(f) => logger.error("Couldn't create Kafka sensor.", f)
+            case Failure(f) => logger.error(s"Couldn't create Kafka sensor for sensor (#${sensor.id}).", f)
+          }
+        case sensor if sensor.sensorType == SensorTypes.Time =>
+          Try(TimeSensor(eventProcessor.eventProcessor, sensor.properties, executionContext, quartzScheduler)) match {
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/sensors/Sensors.scala]: This is fine, but I would expect quartzScheduler to be bundled in za.co.absa.hyperdrive.trigger.scheduler.sensors.time with all dependencies related to time sensor. (this also include src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/Application.scala)

Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/sensors/Sensors.scala]: Ok I agree. Then we need less spring-boot-starter "magic"

File: src/test/scala/za/co/absa/hyperdrive/trigger/scheduler/EventProcessorTest.scala [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+  val eventRepository = mock[EventRepository]
+  val dagDefinitionRepository = mock[DagDefinitionRepository]
+  val dagInstanceRepository = mock[DagInstanceRepository]
+
+  "EventProcessor.eventProcessor" should "persist dag instances for events" in {
+    // given
+
+
+
```
Comment (jozefbakus) [inline src/test/scala/za/co/absa/hyperdrive/trigger/scheduler/EventProcessorTest.scala]: Empty lines

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/sensors/Sensors.scala (hunk 2 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/sensors/Sensors.scala]: It has been changed, but quartzScheduler is still injected in the constructor of this class. I would move it to TimeSensor. And consider not to overload constructor.

File: src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/JobScheduler.scala [hunk 4]
```diff
@@ -48,6 +49,7 @@ class JobScheduler @Inject()(sensors: Sensors, executors: Executors, dagInstance
 
   def startManager(): Unit = {
     if(!isManagerRunningAtomic.get() && runningScheduler.isCompleted) {
+      TimeSensorQuartzSchedulerManager.start()
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/JobScheduler.scala]: Last thing, let's move this to Sensors to a method that will start all required dep. and here will be used only that method.

**Review by jozefbakus — APPROVED** [review]
Nice. Let's merge it.

**Pull request merged**